packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -204,14 +100,54 @@ export class SummaryManager extends EventEmitter implements IDisposable {
             this.setClientId(context.clientId);
         }
 
-        context.quorum.on("addMember", (clientId: string, details: ISequencedClient) => {
+        // Track ops until first (write) connect
+        const opsUntilFirstConnectHandler = (clientId: string, details: ISequencedClient) => {
             if (this.opsUntilFirstConnect === -1 && clientId === this.clientId) {
+                context.quorum.off("addMember", opsUntilFirstConnectHandler);
                 this.opsUntilFirstConnect = details.sequenceNumber - this.context.deltaManager.initialSequenceNumber;
             }
+        };
+        context.quorum.on("addMember", opsUntilFirstConnectHandler);
+
+        this.summaryCollection.on("default", (op) => {
+            const opsSinceLastAckForClient = op.sequenceNumber - this.lastSummaryAckSeqForClient;
+            if (
+                opsSinceLastAckForClient > this.maxOpsSinceLastSummary
+                && !this.hasLoggedTelemetry
+                && this.electedClientId !== undefined
+            ) {
+                // Limit telemetry to only next client?
+                this.logger.sendErrorEvent({
+                    eventName: "ElectedClientNotSummarizing",
+                    thisClientId: this.clientId,
+                    electedClientId: this.electedClientId,
+                    sequenceNumber: op.sequenceNumber,
+                    lastSummaryAckSeqForClient: this.lastSummaryAckSeqForClient,
+                });
+
+                // In future we will change the elected client.
+                // this.orderedClients.incrementCurrentClient();
+            }
+        });
+        this.summaryCollection.on(MessageType.SummaryAck, (op) => {
```

Comment:
do we want to track acks, or just ops? I've gone back and forth a bit on this. if one client's ops fail to ack, do we think another client's will?

Reply:
Acks is more important, but both can be valuable: each summarizer client tracks both for itself- attempt vs success. And no, it's not true that summary attempts that are nacked by one client imply that they will definitely be nacked by another client, especially if the internal state tracking of that summarizer client is busted. Though, in most cases simply restarting the client would be sufficient.